RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc

```diff
@@ -70,13 +71,36 @@ void PatternRecognitionbyCA::makeTracksters(const PatternRecognitionAlgoBase::In
   //#ifdef FP_DEBUG
   const auto &doublets = theGraph_->getAllDoublets();
   int tracksterId = 0;
+
   for (auto const &ntuplet : foundNtuplets) {
     std::set<unsigned int> effective_cluster_idx;
+    std::pair<std::set<unsigned int>::iterator, bool> retVal;
+
+    std::vector<float> times;
+    std::vector<float> timeErrors;
+
     for (auto const &doublet : ntuplet) {
       auto innerCluster = doublets[doublet].innerClusterId();
       auto outerCluster = doublets[doublet].outerClusterId();
-      effective_cluster_idx.insert(innerCluster);
-      effective_cluster_idx.insert(outerCluster);
+
+      retVal = effective_cluster_idx.insert(innerCluster);
+      if (retVal.second) {
+        float time = input.layerClustersTime.get(innerCluster).first;
+        if (time > -99) {
+          times.push_back(time);
+          timeErrors.push_back(1. / pow(input.layerClustersTime.get(innerCluster).second, 2));
+        }
+      }
+
+      retVal = effective_cluster_idx.insert(outerCluster);
+      if (retVal.second) {
+        float time = input.layerClustersTime.get(outerCluster).first;
+        if (time > -99) {
+          times.push_back(input.layerClustersTime.get(outerCluster).first);
```

Comment:
Maybe for consistency use `time` also here?

Reply:
yes sure!